### ISSUE TITLE
[MU4 Issue] "I" shortcut for instruments window does not work

### ISSUE THREAD

**Issue by MarcSabatella** [musescore/MuseScore#10484]
I'm on Linux, but I suspect this is not platform independent, and related more to the change in the implementation of the Instruemnts window from a dialog to a tabbed window.

"I" has always been the shortcut to open this window, and it still shows in the shortcuts list.  But, it isn't hooked up to the current tabbed window; it doesn't nothing.

This is a fairly significant usability issue, in that it is now impossible to close the side panel via keyboard alone.  In previous versions, F8 closed the Inspector and immediately gave more space for the score; F9 then closed the palettes and provided even more more.  So in two keystrokes you could have the full screen width available.  Now, with the new tabbed layout, F8 followed by F9 leaves you with the Instruments window open.

With that in mind, frankly, I'd just as soon have a new shortcut to toggle (or collapse) the entire side pane, rather than requiring three separate keystrokes, especially three keystrokes that aren't adjacent.

**Comment by njvdberg** [comment]
I'm on Linux too but I can't reproduce this issue. Pressing "I" does pop up the instruments window. Also F8 and F9 work as expected.
However I must admit, I'm not using a nightly build but my own build (debug) version.

**Comment by MarcSabatella** [comment]
OK, and now it is bringing up the separate window for me as well.  But, it's the separate window, not the Instruments tab in the left sidebar.  So, still no easy way to close that via keyboard, unless I'm missing something.

**Comment by RobFog** [comment]
I would argue simple letters shouldn't be used as shortcuts without additional keys such as ctrl. They are too easily clicked.

I know many people who were confused about what happened when they first accidentally pressed "m", the keyboard shortcut for multi-bar rests MuseScore 3 and earlier.

**Comment by MarcSabatella** [comment]
I agree having a single-key shortcut for "M" was a bad idea, since most people should almost never need to press it all given they are enabled in parts by default.  Also adding or removing instruments to a score hardly the most common operation, not after you're more than a few minutes into it.  But I couldn't bear for note input mode not to be toggled by a simple key, or need to use a modifier to enter the durations and pitches.

Probably there should be a big rethink of the default shortcuts, but that would put off a lot of existing users.  So a "legacy" shortcut set and a "recommended" one could be nice.

**Comment by RobFog** [comment]
> I agree having a single-key shortcut for "M" was a bad idea, since most people should almost never need to press it all given they are enabled in parts by default. Also adding or removing instruments to a score hardly the most common operation, not after you're more than a few minutes into it.
Agreed.

>But I couldn't bear for note input mode not to be toggled by a simple key, or need to use a modifier to enter the durations and pitches.
Yeah, those are good exceptions to the rule. Activating note input mode doesn't change the score so that's fine. For durations and pitches it's obvious enough to see what is happening when pressing the keys.

> Probably there should be a big rethink of the default shortcuts, but that would put off a lot of existing users. So a "legacy" shortcut set and a "recommended" one could be nice.
Yep.

**Comment by Tantacrul** [comment]
I think it would be nice to assign one key for 'hide side panels' and it would just hide all of them. @bkunda: you're more up on the shortcuts than I am. Any suggestions? I feel we should include this if possible.

**Comment by bkunda** [comment]
Pressing `I` works as I would expect it to (from an MS3 POV): it triggers the **instruments _dialog_** (indeed, the equivalent one to that in MS3). I would not confuse this with the **instruments _panel_**.

But given that the **Palettes** and **Properties** panels each have their own legacy shortcuts for toggling visibility, I do think it makes sense to also something similar for the **instruments** _**panel**_.

`F7` currently doesn't seem to trigger anything in MacOS or Windows. Could we perhaps use this, particularly since it lies adjacent to `F8` (**Properties**) and `F9` (**Palettes**)?

**Comment by Tantacrul** [comment]
I don't think the individual toggles are massively useful if they leave other ones open. I think a much more sensible shortcut would be one that hides/shows all sidebars. In fact, I'd suggest that it is a primary shortcut - possibly not an 'F' number.

So, perhaps F7 can be used for the Instruments panel, like you suggested.

Then and then another shortcut to hide all side panels (and then turn them back on). Which shouldn't affect positioning at all. - Perhaps this needs to be bumped

**Comment by bkunda** [comment]
I get it – a bit like the 'Show/hide UI' feature in Figma. Would be useful for seeing an entire score without the panels in the way. 

I might suggest this is a feature for 4.x, because it's adding something genuinely new, and we'd need to test it in a range of potentially complex situations (undocked panels, panels on different screens, resized panels, etc.). My suggestion of adding a shortcut only for the instruments panel is really to bring its visibility functionality up to speed with the palettes and properties (it's weird the "instruments" doesn't have this and looks like an oversight).

**Comment by Tantacrul** [comment]
Agreed. Cheers!

**Comment by MarcSabatella** [comment]
Since I posted the original issue, I'd add that I'm happy with simply adding a shortcut to toggle the instruments panel for now also. 
 And indeed, eventually a single shortcut to open/close them together would be nice too.  But I expect I'll actually be leaving the instruments panel closed most of the time, so really it's just two keystrokes F8 & F9 to toggle the properties and palettes to get the same basic effect as toggling the whole sidebar.  Not a big deal.

Also, to be clear: when I originally posted this issue, the shortcut "I" didn't open the instruments window - it did *nothing*.  So, that much was already fixed sometime in the last few months.  Also, FWIW, I was under the mistaken assumption that the new instruments panel was designed to *replace* the instruments window.

**Comment by bkunda** [comment]
Closing as new task now raised
